functional-tests.sh

```diff
@@ -96,1139 +96,1083 @@ MC_CONFIG_DIR="/tmp/.mc-$RANDOM"
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd admin user enable "$SERVER_ALIAS" "$username"
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cat "${test_alias}/${BUCKET_NAME}/${object1_name}"
 
-    # check that user can be removed, and then is no longer available
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd admin user remove "$SERVER_ALIAS" "$username"
-    assert_failure "$start_time" "${FUNCNAME[0]}" mc_cmd cat "${test_alias}/${BUCKET_NAME}/${object1_name}"
+	# check that user can be removed, and then is no longer available
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd admin user remove "$SERVER_ALIAS" "$username"
+	assert_failure "$start_time" "${FUNCNAME[0]}" mc_cmd cat "${test_alias}/${BUCKET_NAME}/${object1_name}"
 
-    log_success "$start_time" "${FUNCNAME[0]}"
+	log_success "$start_time" "${FUNCNAME[0]}"
 }
 
-function test_bucket_replication()
-{
-    show "${FUNCNAME[0]}"
-
-    start_time=$(get_time)
-    object_name="mc-test-replicate-$RANDOM"
-    bkt1="rep-bucket1"
-    bkt2="rep-bucket2"
-
-    echo "${SERVER_ALIAS}"
-    mc admin info "${SERVER_ALIAS}"
-    mc ls "${SERVER_ALIAS}"
-    # Remove any pre-existing buckets
-    mc rb "${SERVER_ALIAS}/${bkt1}" --force
-    mc rb "${SERVER_ALIAS}/${bkt2}" --force
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bkt1}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd version enable "${SERVER_ALIAS}/${bkt1}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bkt2}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd version enable "${SERVER_ALIAS}/${bkt2}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd replicate add "${SERVER_ALIAS}/${bkt1}" --remote-bucket "${SERVER_ALIAS}/${bkt2}"
-    loop_count=0
-    while true; do
-        if [ $loop_count -eq 100 ]; then
-            break
-        fi
-        # Intentionally not using mc_cmd to avoid delays of adding error to log files etc
-        assert_success "$start_time" "${FUNCNAME[0]}" mc cp "$FILE_0_B" "${SERVER_ALIAS}/${bkt1}/${object_name}"
-        assert_success "$start_time" "${FUNCNAME[0]}" mc rm "${SERVER_ALIAS}/${bkt1}/${object_name}"
-        assert_failure "$start_time" "${FUNCNAME[0]}" mc stat "${SERVER_ALIAS}/${bkt1}/${object_name}"
-        loop_count=$((loop_count + 1))
-        echo $loop_count
-    done
-    # Finally remove the buckets
-    mc rb "${SERVER_ALIAS}/${bkt1}" --force
-    mc rb "${SERVER_ALIAS}/${bkt2}" --force
-
-    log_success "$start_time" "${FUNCNAME[0]}"
+function test_bucket_replication() {
+	show "${FUNCNAME[0]}"
+
+	start_time=$(get_time)
+	object_name="mc-test-replicate-$RANDOM"
+	bkt1="rep-bucket1"
+	bkt2="rep-bucket2"
+
+	echo "${SERVER_ALIAS}"
+	mc_cmd admin info "${SERVER_ALIAS}"
+	mc_cmd ls "${SERVER_ALIAS}"
+	# Remove any pre-existing buckets
+	mc_cmd rb "${SERVER_ALIAS}/${bkt1}" --force
+	mc_cmd rb "${SERVER_ALIAS}/${bkt2}" --force
```

Comment:
Would these fail if we try removing non existing buckets?

Reply:
thats for a different PR to deal with :-) @shtripat